pkg/sql/authorization.go

```diff
@@ -451,3 +463,29 @@ const ConnAuditingClusterSettingName = "server.auth_log.sql_connections.enabled"
 // counts in SetClusterSetting() and importing pgwire here would
 // create a circular dependency.
 const AuthAuditingClusterSettingName = "server.auth_log.sql_sessions.enabled"
+
+func (p *planner) canCreateOnSchema(
+	ctx context.Context, dbID descpb.ID, schemaID descpb.ID, user string,
+) error {
+	resolvedSchema, err := resolver.ResolveSchemaDescByID(ctx, p.Txn(), p.ExecCfg().Codec, schemaID)
+	if err != nil {
+		return err
+	}
+
+	switch resolvedSchema.Kind {
+	case sqlbase.SchemaPublic:
+		// Anyone can CREATE on a public schema.
+		return nil
+	case sqlbase.SchemaTemporary, sqlbase.SchemaVirtual:
```

Comment:
I think that the virtual schema check should be its own case always returning an error

Reply:
Would it be fine to have the same error message logic for SchemaTemporary and SchemaVirtual? 
Since these cases right now will both just lookup the schema name and return an error for it.

And looking up the schema name for a virtual schema should be okay.

Reply:
yeah you're right, I think i read this wrong initially. However, I am making a change in https://github.com/cockroachdb/cockroach/pull/52726 that you might want to just copy. I adjusted the `ResolvedSchema` struct to always contain the name of the resolved schema, so that you don't have to look it up again here.

Reply:
Done